### PR TITLE
combined BGC data timestamp

### PR DESCRIPTION
**Pull request by leonardolaiolo**
Bit of a repetition in the commits due to a mistake on tagging the right issue - sorry. I tried to make it cleaner, but commits just got repeated. I hope it is not too confusing, please come around or ask me if something does not make sense.

This should fix both these issues:
- https://github.com/aodn/public-schema/issues/53
- https://github.com/aodn/public-schema/issues/60

Changes could have been done in different ways, I decided to change every single product instead of modifying the `combined_bgc_map`. I thought we could make Claire happy this way, as she asked in a recent email if the repetition of `trip_code` and `Trip_Code` could be avoided in the final products. Furthermore, because of the order of parameters, hard coding for each products could have been required anyway, as we now have different `SampleTime_Local` for each products and combined product (i.e., `SELECT bm.*` would not work) - and we do need to have  `SampleTime_Local` in the `combined_bgc_map` otherwise Step 2 in AODN Portal breaks.

Geoserver needs this PR merged once this is on master (impacts filter) https://github.com/aodn/geoserver-config/pull/744

### CODE REVIEW

File: bgc_data/combined_bgc_data.sql [hunk 1]
```diff
@@ -54,7 +56,16 @@ UNION
...
 )
    SELECT
-      bm.*,
+      bm."Project",
+      bm."StationName",
+      bm."TripCode",
+      CASE WHEN che."SampleTime_Local" IS NOT NULL THEN che."SampleTime_Local"
+           WHEN (che."SampleTime_Local" IS NULL) AND (pig."SampleTime_Local" IS NULL) AND (tss."SampleTime_Local" IS NULL) THEN ppl."SampleTime_Local"
+           WHEN (che."SampleTime_Local" IS NULL AND pig."SampleTime_Local" IS NULL) THEN tss."SampleTime_Local"
+           WHEN che."SampleTime_Local" IS NULL THEN pig."SampleTime_Local" END AS "SampleTime_Local",
```
Comment (mhidas) [inline bgc_data/combined_bgc_data.sql:65]: Remember the `COALESCE` function!
```suggestion
    COALESCE(che."SampleTime_Local", pig."SampleTime_Local", tss."SampleTime_Local", ppl."SampleTime_Local") AS "SampleTime_Local",
```

File: bgc_data/bgc_chemistry_data.sql [hunk 2]
```diff
@@ -2,7 +2,13 @@
 --includes metadata
 CREATE MATERIALIZED VIEW bgc_chemistry_data AS
    SELECT
-      bm.*,
+      bm."Project",
+      bm."StationName",
+      bm."TripCode",
+      che.sampledatelocal AS "SampleTime_Local",
```
Comment (mhidas) [inline bgc_data/bgc_chemistry_data.sql:8]: Remember we need to format local times to a string, e.g.
https://github.com/aodn/public-schema/blob/821454fd289d6f9818af9f060ccc125c55d02627/bgc_data/bgc_trip_metadata.sql#L11

Comment (mhidas) [reply bgc_data/bgc_chemistry_data.sql:8]: I seem to remember Geoserver is still able to filter on these string fields as if they were timestamps, but would be good to re-confirm that.

File: bgc_data/combined_bgc_map.sql [hunk 3]
```diff
@@ -5,8 +5,8 @@ CREATE MATERIALIZED VIEW combined_bgc_map AS
     "Project",
     "StationName",
     "TripCode",
-    "SampleTime_Local"::date AS "SampleDate_Local",
-    "SampleTime_UTC"::date AS "SampleDate_UTC",
+    "SampleTime_Local",
```
Comment (mhidas) [inline bgc_data/combined_bgc_map.sql:8]: This `SampleTime_Local` will now not match any field of the same name in any of the `_data` layers. Might lead to some confusion, especially if we use this field to filter on in the portal?

Comment (mhidas) [reply bgc_data/combined_bgc_map.sql:8]: My suggestion would be to rename this to `TripTime_Local` (and also have `TripTime_UTC`) and include it in all the `_data` views - then we can filter on that.
@clairedavies what do you think?

**Review by mhidas — COMMENTED** [review]
Looks good, but I have suggested a few tweaks.

File: bgc_data/bgc_chemistry_data.sql (hunk 2 above)
Comment (leonardolaiolo) [reply bgc_data/bgc_chemistry_data.sql]: done in commit [ce7a967](https://github.com/aodn/public-schema/pull/59/commits/ce7a9671b041df029d470e6c655a6337df20fb7d)

**Review by ana-berger — APPROVED** [review]

**Pull request merged**